In [5]:
import pandas as pd
import numpy as np
import matplotlib as plt
import seaborn as sns

train = pd.read_csv('../data/train.csv')
test  = pd.read_csv('../data/test.csv')

print(f"Train: {train.shape[0]} rows, {train.shape[1]} columns")
print(f"Test: {test.shape[0]} rows, {test.shape[1]} columns")

print()

print('Missing values in train:')
print(train.isnull().sum()[train.isnull().sum() > 0])


Train: 891 rows, 12 columns
Test: 418 rows, 11 columns

Missing values in train:
Age         177
Cabin       687
Embarked      2
dtype: int64


# Results

## 891 passengers in train, 418 in test — this is the classic Kaggle Titanic split
## Test has 11 columns instead of 12 because it's missing Survived — that's the column we're going to predict
## Three problems to fix: Age (177 missing), Cabin (687 missing — that's 77%!), Embarked (just 2 missing)

In [ ]:
# This function takes a full name and pulls out just the title
# Example: "Braund, Mr. Owen Harris" → splits on ',' → " Mr. Owen Harris"
#          then splits on '.' → " Mr" → strip() removes spaces → "Mr"

def extract_title(name):
    return name.split(',')[1]